src/url.js

```diff
@@ -0,0 +1,70 @@
+const FlowdockText = require('flowdock-text');
+
+function createTextNode(state, content, level) {
+  var token = new state.Token('text', '', 0);
+  token.content = content;
+  token.level = level;
+  return token;
+}
+
+function flowdockUrl(md, state) {
+  const tokens = state.tokens;
+  const arrayReplaceAt = md.utils.arrayReplaceAt;
+  
+  var nodes = [];
+  var token, children;
+
+  for (var i = 0; i < tokens.length; i++) {
+    if (tokens[i].type !== 'inline') { continue; }
+    
+    children = tokens[i].children;
+
+    for (var j = children.length -1; j >= 0; j--) {
+      if(children[j].type !== 'text') { continue; }
+
+      const childContent = children[j].content;
+      const childLevel = children[j].level;
+
+      const urlPart = FlowdockText.extractUrlsWithIndices(childContent);
+      var endOfLastUrl = 0;
+      for (var k = 0; k < urlPart.length; k++) {
+        const url = urlPart[k];
+
+        if (url.indices[0] > 0) {
+          nodes.push(createTextNode(state, childContent.slice(endOfLastUrl, url.indices[0]), childLevel));
+        }
+
+        token       = new state.Token('link_open', 'a', 1);
+        token.attrs = [ [ 'href', url.url ]];
+        token.level = childLevel;
+        nodes.push(token);
+
+        token         = new state.Token('text', '', 0);
+        token.content = url.url;
+        token.level   = childLevel;
+        nodes.push(token);
+
+        token       = new state.Token('link_close', 'a', -1);
+        token.level = childLevel;
+        nodes.push(token);
+
```

Comment:
I might try to clean up this code a bit with helper procs for creating the various token types, it might read a little cleaner then:

    nodes.push(createTokenOpen(url, childLevel));
    nodes.push(createTokenText(url, childLevel));
    nodes.push(createTokenClose(childLevel));